### DISCUSSION TITLE
Skip days should better interact with "every N days" habits

### DISCUSSION THREAD

**Discussion by foonathan** [iSoron/uhabits#1686] [Feature Requests]
Maybe I've just misunderstood the purpose of skip days, but I find their interaction with "every N days" habits a bit confusing/not useful.

Let's say I want to go to the gym every 3 days. On day 1 I go to the gym, which also marks days 2 and 3 as completed. Then I go again on day 4, so far so good. However, then I go on vacation days 5-7 where I cannot go to the gym, so I skip those days. I could go to the gym on day 8, but don't, I only go on day 9. Since 8 - 4 > 3, day 8 is left gray and resets the streak.

I did not expect this behavior. Since I've skipped 3 days, I've expected them to be completely skipped for the "every 3 days" rule, as if they never existed in the first place. If this behavior is intended, I'd like to see an alternative to specify a "vacation" or something with the semantics I've expected.

**Comment by texnic** [comment]
First of all, big kudos to the main developer and all contributors!  I love this app, and it actually helps developing good habits.

I am facing an issue with skipping in non-everyday habits as well. It's actually quite interesting. I tested it for the sake of this report.

I have a "6 times per week" habit.  I have reached a 100% score on it.
Then I skipped 4 days (not missed, but skipped).
In case it matters: I am now at "total" of 218. 
Depending on how many days passed since the skipping days, the score changes.
Today is 28 April.
If the skipped days are 25-28 April, the score remains 100%, as expected.
Skipped 24-27 April, habit done again on 28 April; score 99%.
Skipped 23-27 April, done again on 27-28 April; score 98%.
Skipped 22-25 April, done again on 26-28 April; score 98%.
Skipped 21-24 April, done again on 25-28 April; score 97%. 
And so on. If the skipped days are long in the past, today's score starts increasing again.

The expected behaviour is that the score remains at 100 % in all cases above, since the "skips keep the score unchanged and don't break the streak" (by definition).

The counterintuitive result of the above is that, in the score graph, the score remains stable as long as I am skipping days but starts _decreasing_ once I start _fulfilling_ the habit after a few skipped days; see image. 

![Screenshot_20230428_110009](https://user-images.githubusercontent.com/6185463/235104422-76d23379-8ca9-4ed1-abee-56003017b09b.jpg)

**Reply by frrk** [reply]
I have the same issue. If I wear jeans 3 days in a row, on day 4 I want to swap them for clean ones. If it is a very hot day, I will wear shorts instead and thus skip a day.

Now what happens is that it will still remind me to change my jeans after 3 days, even if I skipped days 2 and 3.
